lib/chef/resource/powershell_script.rb

```diff
@@ -22,6 +22,21 @@ class Resource
     class PowershellScript < Chef::Resource::WindowsScript
       provides :powershell_script, os: "windows"
 
+      property :flags, String,
+        description: "A string that is passed to the Windows PowerShell command",
+        default: lazy { default_flags },
+        coerce: proc { |input|
+          if input == default_flags
+            # Means there was no input provided,
+            # and should use defaults in this case
+            input
+          else
+            # The last occurance of a flag would override its
+            # previous one at the time of command execution.
+            [default_flags, input].join(" ")
+          end
```

Comment:
you should be able to just `proc { ( default_flags + input ).join(" ") }` you might need to have an `input ||= []` as well to make sure input is always an Array.  come to think if it you might want to `input.split(/\s+/) if input.is_a?(String)`

Reply:
Right sir, but as per the [Document](https://docs.chef.io/resource_powershell_script.html), which was corrected after [this discussion](https://github.com/chef/chef/issues/1604#issuecomment-78310266), `flags` are of String type and not Array. Even the multiple flags are also passed within the same string simply separated by space.